custom-metrics-stackdriver-adapter/pkg/apiserver/installer/apiserver_test.go

```diff
@@ -127,12 +127,10 @@ func handle(prov provider.CustomMetricsProvider) http.Handler {
 
 			Context:                reqContextMapper,
 			OptionsExternalVersion: &schema.GroupVersion{Version: "v1"},
-
-			ResourceLister: provider.NewResourceLister(prov),
 		},
 	}
 
-	if err := group.InstallREST(container); err != nil {
+	if err := group.InstallREST(container, prov, reqContextMapper); err != nil {
```

Comment:
without having an overview about your code structure, the creation of another request context mapper looks suspicious. It's a global state and having a second one is usually wrong.

Reply:
I'm not familiar with the logic that creates reqContextMapper, can we leave it as it is? If this is a bug, I think it should be fixed here (where this code is forked from): https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/pkg/apiserver/installer/apiserver_test.go

Reply:
It think is should be fixed in both places.
cc @DirectXMan12

Reply:
I agree, IF it is a bug. I think it's not, it looks to me that the request context mapper is created only here for testing purposes.

What I mean is, we don't need to block this PR on this, this is unrelated to the changes I'm making.

Reply:
right, in tests this can be ok.